### PR TITLE
bpo-32173: linecache.py add lazycache to __all__ and use dict.clear to clear the cache

### PR DESCRIPTION
**Pull request by ganziqim**
https://bugs.python.org/issue32173

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Review by lisroach — APPROVED** [review]
I think this is a reasonable diff, lazycache should be added to __all__ and the clearcache is improved by using clear().

The extra code changes are just semantics which help code readability without changing any functionality.

**Comment by GanZiQim** [comment]
I've been signed for three times, but the state has not changed

**Comment by asvetlov** [comment]
CLA status updated

**Comment by GanZiQim** [comment]
news entry added

**Review by asvetlov — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2017-12-04-10-14-23.bpo-32173.e0C5dF.rst
Comment (Mariatta) [inline Misc/NEWS.d/next/Library/2017-12-04-10-14-23.bpo-32173.e0C5dF.rst]: The news entry should be improved. The bpo issue title is not always appropriate as a News entry.

Your news says adding lazycache and use dict.clear to clear the cache, but this patch actually does a little more than those two.

Use reST markup, and end the sentence with a period.

**Review by Mariatta — CHANGES_REQUESTED** [review]

**Comment by csabella** [comment]
This issue is approved by 2 core developers, but was waiting on an update to the blurb.

@ganziqim Please make the last changes requested by Mariatta, after which I believe this will be ready to merge.  Thanks!

**Comment by ganziqim** [comment]
> This issue is approved by 2 core developers, but was waiting on an update to the blurb.
> 
> @ganziqim Please make the last changes requested by Mariatta, after which I believe this will be ready to merge. Thanks!

done.

**Comment by csabella** [comment]
@ganziqim Thanks!

@lisroach @asvetlov I believe Mariatta is out of open source starting today, so, if you can, please review the changes made to the news entry.  Thank you!

**Pull request merged**
